### DISCUSSION TITLE
[Feature] Is it posible to set two or more server urls in app?

### DISCUSSION THREAD

**Discussion by zsimple** [immich-app/immich#11023] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I have tried photoprism, LibrePhotos photo management tools, and finally chose Immich, which is really great!

My use case is, I have deployed the Immich server on my home NAS, and then mapped it to the internet through a tunnel. When I'm connected to the home wifi, I use the local address on immich app, which is very fast and free. When I'm outside, I have to switch to the internet address, which is slower and I have to pay for it. Every time I switch, I have to exit the app and re-enter the address and log in again, which is very inconvenient.

It is very hard to handle it with dns, for internet address is https ( https://immich.example.com for example), while home address is http and different port (http://192.168.0.10:8080, it's a docker container).

So, can the app maintain two addresses and automatically select the appropriate server address based on the network, auto-speed test, or priority? It will be very helpful. I think many people will like it, or is there a better way to handle this case ?

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by zsimple** [comment]
Sorry I seached using other words and found someone have already discussed about this, but I cant agree with the solution, as I said, it is unpossible to hand it by split dns, and it is also to hard for normal users.

Anyway, closing it.

**Reply by johnmmcgee** [reply]
You would handle this with split dns.  As for the port, you could handle that with a reverse proxy.

**Reply by zsimple** [reply]
Thanks, but I think port is not the big thing, the main difficult is different schema(http and https).

I tried to create a self-signed cert, and install root ca in my phone, I can visit the https by browser without any problem, but it seems that the immich app reject my self-signed https access.

**Reply by bo0tzz** [reply]
You would do split dns on your domain name, so that inside your network it resolves to the local address of your reverse proxy.

**Comment by Hritik14** [comment]
For posteriority, the iOS app has this feature now !

**Reply by maxint137** [reply]
Thank you @Hritik14 ! Can you please share a reference to learn more about that feature?

**Reply by Hritik14** [reply]
It's at settings -> networking inside the app. Not sure where it's documented.
